### PR TITLE
fix player kicked for flying with jetpack

### PR DESCRIPTION
**Pull request by ThatGamerBlue**
sorry about all the refactoring intellij likes to do it without me allowing it to

### CODE REVIEW

**Comment by thommy101** [comment]
Tip: crtl + alt + s -> Editor -> Codestyle -> Java -> Imports -> Class count to use import with '*'. Change that one to a higher value to prevent auto merging of imports.

**Comment by ThatGamerBlue** [comment]
@thommy101 omg thank you <3

File: src/main/java/mekanism/common/CommonEventHandler.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * @author ThatGamerBlue
+ */
+public class CommonEventHandler {
+
+    private boolean canFly = false;
+
+    @EventHandler
+    public void onPlayerTick(TickEvent.PlayerTickEvent event){
+        if(event.phase == TickEvent.Phase.START && event.side == Side.SERVER){
+            if(canFly){
```
Comment (Bluexin) [inline src/main/java/mekanism/common/CommonEventHandler.java]: Seems like you cache a value on the serverside.
One for all players, while this event is called for each player.
I'm highly doubtful about this working properly.

File: src/main/java/mekanism/common/CommonEventHandler.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+                if(!(event.player.inventory.armorInventory[1].getItem() instanceof ItemJetpack)){
+                    canFly = false;
+                    event.player.capabilities.allowFlying = canFly;
```
Comment (Bluexin) [inline src/main/java/mekanism/common/CommonEventHandler.java]: Basing it solely on mek jetpack would probably break any other mod changing this wouldn't it?

Comment (ThatGamerBlue) [inline src/main/java/mekanism/common/CommonEventHandler.java]: thats their problem

Comment (Bluexin) [inline src/main/java/mekanism/common/CommonEventHandler.java]: I do not like that idea, because it'd go against good modding practices.

**Comment by Tencao** [comment]
I second what has been said here. There needs to be stricter checks as to when to allow flying, rather than just checking if the item is in the inventory.  Right now, you're not even checking if the item has enough fuel to even fly, you're just saying if the player has the item. I have several mods in my modpack that control flying to some degree, and having the mentality of "its their problem" is not a fix at all, its laziness.

**Comment by ThatGamerBlue** [comment]
the allowFlight player capability doesnt mean the player can fly, it means if the player is flying by whatever means the player wont get kicked

**Comment by thommy101** [comment]
The way you've implemented it, when the player stops wearing the jetpack, you set the flying capability to false, but what if an other mod has set it to true? That's overwritten with your false at that moment.

**Comment by Bluexin** [comment]
Yeah, but according to some people it's their problem ¯\_(ツ)_/¯

**Comment by aidancbrady** [comment]
Not a big fan of modifying flight capabilities, sorry.

**Pull request closed without merging**